status/status.go

```diff
@@ -116,13 +121,14 @@ type Status struct {
 
 func newStatus() *Status {
 	return &Status{
-		ImageCount:   MapMapCount{},
-		PhaseCount:   MapMapCount{},
-		PodState:     PodStateMap{},
-		PodsCount:    PodsCount{},
-		CiliumStatus: CiliumStatusMap{},
-		Errors:       ErrorCountMapMap{},
-		mutex:        &sync.Mutex{},
+		ImageCount:      MapMapCount{},
+		PhaseCount:      MapMapCount{},
+		PodState:        PodStateMap{},
+		PodsCount:       PodsCount{},
+		CiliumStatus:    CiliumStatusMap{},
+		CiliumEndpoints: CiliumEndpointsMap{},
```

Comment:
Yes, that has been my initial attempt, but unfortunately it appears that `cilium status` doesn't currently expose any information about the termination of endpoints regeneration. I was on the fence about whether to introduce that on the agent or implement the functionality in the CLI, and at the end I preferred the latter mainly for convenience, given that it work with any version. Extending the status instead would only apply to the latest Cilium version, or require to be backported to stable branches with the associated risks/complexity. I don't have a strong opinion though, and I'm open to switch to the other approach that if there's consensus on it.

Reply:
I raised similar concerns in [another PR](https://github.com/cilium/cilium-cli/pull/2212#discussion_r1446119226). Even though the main discussion was about using a structured output (JSON instead of text) - the aspect of having to implement and backport the functionality in the Cilium Agent was the same. 

With the same argument that backporting of new functionality should be straightforward and a non-blocker. Especially if it helps to prevent flakes in CI on these stable branches too.

Even though slightly different, I though it's fair to raise this point here as well. what do the maintainers think about this? (@tklauser @michi-covalent )

Reply:
I thought a bit more about this, and the main thing that to me feels weird about adding it to the general status is that we would have a separate `Endpoint` section with only the `RestorationCompleted` information, while all the rest of endpoint-related information can be retrieved through `cilium endpoint list`. Backporting such change would be also likely easy, but not necessarily trivial (in the sense of zero conflicts), as there have been a few changes over time in terms of how we track the completion of the regeneration of endpoints.

WDYT? Does it sound reasonable to you to keep the current CLI-only approach?

Reply:
IMO, the one (having a status for a subsystem - even if it's just the first status-info of this subsystem (restoration completed)) should not exclude the other (retrieving detailed information from / interacting with the subsystem) :shrug: 

The status should provide an initial overview of the system with its subsystems (and outline potential issues)- and not replace other commands (that are more detailed) - that are used to further investigate potential issues.

Reply:
In general I agree with the point raised by @mhofstetter but I think in this particular case the costs of changing `cilium-dbg` and backporting to all stable branches doesn't justify the cleaner implementation. I'd tend to say that it's fine to keep the current CLI-only approach for the sake of this PR and maybe consider changing to the cleaner approach as a follow-up.